test/mocks/overrides/override_comment_view_model.dart

```diff
@@ -0,0 +1,51 @@
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/ui/comment_post.dart";
+import "package:proxima/viewmodels/comment_view_model.dart";
+
+import "../data/post_comment.dart";
+
+/// A mock implementation of the [CommentViewModel] class.
+/// By default it exposes an empty list of [CommentPost] and does nothing on refresh.
+class MockCommentViewModel
+    extends AutoDisposeFamilyAsyncNotifier<List<CommentPost>, PostIdFirestore>
+    implements CommentViewModel {
+  final Future<List<CommentPost>> Function(PostIdFirestore arg) _build;
+  final Future<void> Function() _onRefresh;
+
+  MockCommentViewModel({
+    Future<List<CommentPost>> Function(PostIdFirestore arg)? build,
+    Future<void> Function()? onRefresh,
+  })  : _build =
+            build ?? ((PostIdFirestore arg) async => List<CommentPost>.empty()),
+        _onRefresh = onRefresh ?? (() async {});
+
+  @override
+  Future<List<CommentPost>> build(PostIdFirestore arg) => _build(arg);
+
+  @override
+  Future<void> refresh() => _onRefresh();
+}
+
+final mockEmptyCommentViewModelOverride = [
+  commentListProvider.overrideWith(
+    () => MockCommentViewModel(),
+  ),
+];
+
+final mockNonEmptyCommentViewModelOverride = [
+  commentListProvider.overrideWith(
+    () => MockCommentViewModel(
+      build: (PostIdFirestore arg) async => testComments,
+    ),
+  ),
+];
+
+final mockLoadingCommentViewModelOverride = [
```

Comment:
Is it normal this varialbe is used nowhere?

Reply:
Yes, that was to expose the main mocks that someone might be interested in when designing a future test.
But I can remove it if you think it would be better like that ?

Reply:
I'm not sure I see in which context it would be needed. Though, it would allow to potentially avoid a merge conflict and improve structure if someone needs it at some point. 
I personally think this piece of code is unnecessary and that unused code is not good, but I'll leave this choice at your discretion.

Reply:
After giving another though about I also didn't found a good context for it's usage.
Thus I'll follow your suggestion :+1: 

Fix in 5124caddd0411991f6d889fe6d019106f2317f5c.